### PR TITLE
DIY quick start improvements

### PR DESCRIPTION
**Pull request by cyrilou242**
Hello, 

This PR adds some resources that helped me get the demo up and running.

### BQ tables 
- `scripts/create_bq_tables.py`:  new script to create the BQ tables, partitioned by ingestion time
- `script/schemas`: Table schemas in json format for consistency and usage with other tools (Terraform, bq mk, BQ api)

### Demo data loading
- ingestion time partition is hardcoded to work with the demo experiment config dates in `example_experiment_config.json`.

### Other
- minor doc fixes
- doc for BQ setup scripts in DOCS.md
- training demo: stop if no data is obtained from BQ to avoid hard to understand error later
- GCP sa credential optional for BQ setup and training scripts. Use default service account by default. GCP related variables can be overridden by passing them as parameters.
- new argument in `workflows/predict.py` to draw a decision from the model scores (exploitation exploration decision). Greedy and softmax ranking.

Really not sure for the minimum python version, I took the one used in the CI. 
Project is recent so the doc about styling, code contributing and all is not there yet, so I did not spent too much time on these points. The code should pass black and should not add errors to flake8, but it may depend on your config of those tools.

### CODE REVIEW

**Comment by econti** [comment]
Thanks for opening this PR! Myself and a teammate are taking a look.

**Comment by joegilley** [comment]
this is outstanding @cyrilou242 ! thanks for your contributions 👍  you're spot on, this project being new we're still establishing our CONTRIBUTING.md file, so really appreciate your thoughtfulness in its absence.

File: DOCS.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 This guide walks through training and serving bandit models.
 
 ## Installation
+banditml requires Python (>= 3.8).
```
Comment (econti) [inline DOCS.md:7]: Change to `>= 3.7`
